server/src/main/java/org/opensearch/gateway/TransportNodesListGatewayStartedShardsBatch.java

```diff
@@ -0,0 +1,458 @@
+            shardIdsWithCustomDataPath = in.readMap(ShardId::new, StreamInput::readString);
+        }
+
+        public Request(DiscoveryNode[] nodes, Map<ShardId, String> shardIdStringMap) {
+            super(nodes);
+            this.shardIdsWithCustomDataPath = Objects.requireNonNull(shardIdStringMap);
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            out.writeMap(shardIdsWithCustomDataPath, (o, k) -> k.writeTo(o), StreamOutput::writeString);
+        }
+
+        public Map<ShardId, String> getShardIdsMap() {
+            return shardIdsWithCustomDataPath;
+        }
+    }
+
+    /**
+     * Responses received from set of other nodes is clubbed into this class and sent back to the caller
+     * of this transport request. Refer {@link TransportNodesAction}
+     *
+     * @opensearch.internal
+     */
+    public static class NodesGatewayStartedShardsBatch extends BaseNodesResponse<NodeGatewayStartedShardsBatch> {
+
+        public NodesGatewayStartedShardsBatch(StreamInput in) throws IOException {
+            super(in);
+        }
+
+        public NodesGatewayStartedShardsBatch(
+            ClusterName clusterName,
+            List<NodeGatewayStartedShardsBatch> nodes,
+            List<FailedNodeException> failures
+        ) {
+            super(clusterName, nodes, failures);
+        }
+
+        @Override
+        protected List<NodeGatewayStartedShardsBatch> readNodesFrom(StreamInput in) throws IOException {
+            return in.readList(NodeGatewayStartedShardsBatch::new);
+        }
+
+        @Override
+        protected void writeNodesTo(StreamOutput out, List<NodeGatewayStartedShardsBatch> nodes) throws IOException {
+            out.writeList(nodes);
+        }
+    }
+
+    /**
+     * NodeRequest class is for deserializing the  request received by this node from other node for this transport action.
+     * This is used in {@link TransportNodesAction}
+     *
+     * @opensearch.internal
+     */
+    public static class NodeRequest extends TransportRequest {
+        private final Map<ShardId, String> shardIdsWithCustomDataPath;
+
+        public NodeRequest(StreamInput in) throws IOException {
+            super(in);
+            shardIdsWithCustomDataPath = in.readMap(ShardId::new, StreamInput::readString);
+        }
+
+        public NodeRequest(Request request) {
+            this.shardIdsWithCustomDataPath = Objects.requireNonNull(request.getShardIdsMap());
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            out.writeMap(shardIdsWithCustomDataPath, (o, k) -> k.writeTo(o), StreamOutput::writeString);
+        }
+
+    }
+
+    /**
+     * Class for storing the information about the shards fetched on the node.
+     *
+     * @opensearch.internal
+     */
+    public static class NodeGatewayStartedShards {
```

Comment:
don't you need to adapt this for batch?

Reply:
We need this as well in batch mode. See NodeGatewayStartedShardsBatch class below which uses this class to keep the Map of ShardId and NodeGatewayStartedShards object